ee/control/consumers/flareconsumer/flareconsumer.go

```diff
@@ -41,23 +38,24 @@ func (f *FlareRunner) RunFlare(ctx context.Context, k types.Knapsack, flareStrea
 	return checkups.RunFlare(ctx, k, flareStream, checkups.InSituEnvironment)
 }
 
-func New(logger log.Logger, knapsack types.Knapsack) *FlareConsumer {
+func New(knapsack types.Knapsack) *FlareConsumer {
 	return &FlareConsumer{
 		flarer:   &FlareRunner{},
 		knapsack: knapsack,
-		logger:   logger,
 		newFlareStream: func(note, uploadRequestURL string) (io.WriteCloser, error) {
 			return shipper.New(knapsack, shipper.WithNote(note), shipper.WithUploadRequestURL(uploadRequestURL))
 		},
 	}
 }
 
 func (fc *FlareConsumer) Do(data io.Reader) error {
+	// slog needs a ctx
```

Comment:
Do we prefer this over the option without a ctx -- `fc.knapsack.Slogger().Info("skipping flare, run too recently"...`?

Reply:
That works for me! If we want to standardize on this, then we can also enforce it via sloglint's [context-only](context-only) setting.